src/Migrations/M005TurnBoxoutSettingIntoBlock.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+use WP_Post;
+
+/**
+ * Update articles block attribute in block innerHTML.
+ */
+class M005TurnBoxoutSettingIntoBlock extends MigrationScript {
+
+	private const BOXOUT_META_KEY = 'p4_take_action_page';
+
+	/**
+	 * Perform the actual migration.
+	 *
+	 * @param MigrationRecord $record Information on the execution, can be used to add logs.
+	 *
+	 * @return void
+	 */
+	protected static function execute( MigrationRecord $record ): void {
+		$args = [
+			'posts_per_page' => - 1,
+			'meta_query'     => [
+				[
+					'key'     => self::BOXOUT_META_KEY,
+					'value'   => '',
+					'compare' => 'NOT IN',
+				],
+			],
+		];
+
+		$tab_posts = get_posts( $args );
+		echo 'Converting ' . count( $tab_posts ) . " posts with Take Action Boxout setting to blocks.\n";
+
+		foreach ( $tab_posts as $tab_post ) {
+			// If the post already has a TAB inside, that one took precedence before and only meta cleanup needed.
+			if ( ! has_block( 'planet4-blocks/take-action-boxout', $tab_post ) ) {
+				self::append_boxout_from_meta( $tab_post );
+			}
+			delete_post_meta( $tab_post->ID, self::BOXOUT_META_KEY );
+		}
+	}
+
+	/**
+	 * Add the Take Action Boxout block that is in the meta to the end of the post content.
+	 *
+	 * @param WP_Post $post A post that has a TAB set through its meta.
+	 */
+	private static function append_boxout_from_meta( WP_Post $post ): void {
+		$attrs  = [
+			'take_action_page' => get_post_meta( $post->ID, self::BOXOUT_META_KEY, true ),
+		];
+		$boxout = '<!-- wp:planet4-blocks/take-action-boxout ' . wp_json_encode( $attrs, JSON_UNESCAPED_SLASHES ) . ' /-->';
```

Comment:
@sagarsdeshmukh Now that the [required changes to the editor](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/dd7e7741658854f32f05465cd50fd8a0d62e0b6f/assets/src/blocks/TakeActionBoxout/TakeActionBoxoutEditor.js#L42-L93) are merged, this migration should have the right result in the editor as well.

Reply:
Hi @Inwerpsel , I created a new post with selecting a dropdown take action boxout page, ran migration script, 
it added a below below -
`<!-- wp:planet4-blocks/take-action-boxout /-->`
block part at the end of page(without the take action page ID). not sure If I miss something 🤔 
could you pelase check and confirm once :)

Reply:
Thanks for spotting Sagar, it was caused by the type the script adding the ID as a string, which wasn't obvious previously because of the other issues with the block that are now resolved.

Interestingly, on the front end it was working even with the string value. In PHP registration uses `integer` for the type, whereas the front end uses `number`. Seems like only `integer` accepts strings.
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/75a1f9be1014796b71425f3b69e93b22952c2cdc/assets/src/blocks/TakeActionBoxout/TakeActionBoxoutBlock.js#L18-L21
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/75a1f9be1014796b71425f3b69e93b22952c2cdc/classes/blocks/class-takeactionboxout.php#L40-L43

Reply:
I updated the script to convert these to an `int`.